BetaCameras/ifm/O3D3xx.cs

```diff
@@ -23,23 +24,48 @@ public class O3D3xx : Camera
         [DllImport("iphlpapi.dll")]
         public static extern int SendARP(int DestIP, int SrcIP, [Out] byte[] pMacAddr, ref int PhyAddrLen);
 
-        private BackgroundWorker updateWorker;
-        private object updateLock = new object();
-        private bool frameAvailable = false;
+        private enum Mode
+        {
+            Run = 0,
+            Edit = 1,
+        }
 
         #region Public Properties
+
+        #region Frequency Channel
         /// <summary>
         /// Frequency channel
         /// </summary>
+        private int _frequencyChannel = 0;
         public int FrequencyChannel
         {
             get
             {
-                return GetFrequencyChannel();
+                if (!this.IsConnected)
+                {
+                    return -1;
+                }
+                SetConfigurationMode(Mode.Edit);
+                _edit.EditApplication(_applicationId);
+
+                _frequencyChannel = Convert.ToInt32(_appImager.GetParameter("Channel"));
+
+                SetConfigurationMode(Mode.Run);
+                return _frequencyChannel;
             }
             set
             {
-                SetFrequencyChannel(value);
+                _frequencyChannel = value;
+                if (!this.IsConnected)
+                {
+                    return;
+                }
+                SetConfigurationMode(Mode.Edit);
+                _edit.EditApplication(_applicationId);
+                _appImager.SetParameter("Channel", value.ToString());
+                _app.Save();
+                _edit.StopEditingApplication();
+                SetConfigurationMode(Mode.Run);
```

Comment:
There seem to be many very similar code-blocks that look like this. And, if something fails, what mode is the object in? For example, one of the `Conver.To...()` calls could throw an exception that should not 'break' the camera instance.

I suggest that you create a utility method that does something like this:

```
private void DoEdit(Action<IEdit> editAction)
{
    if (/* mode not run */) throw InvalidOperation...
    SetConfigurationMode(Mode.Edit);
    try
    {
        editAction(_edit);

        _app.Save();
    }
    finally
    {
        _edit.StopEditingApplication();
        SetConfigurationMode(Mode.Run);
    }
```

Then you can replace these mostly-identical blocks with: `DoEdit((edit) => edit.SetParameter(...))`